File: solidworks-pdm-api/getting-started/add-ins/create/SampleAddIn.cs
using EdmLib;
using System;
using System.Runtime.InteropServices;

namespace CodeStack
{
    [ComVisible(true)]
    [Guid("3A601AFC-7007-46A7-9E71-D3BD41B5E2E2")]
    public class PdmAddInSample : IEdmAddIn5
    {
        const int TEST_CMD_ID = 1;

        public void GetAddInInfo(ref EdmAddInInfo poInfo, IEdmVault5 poVault, IEdmCmdMgr5 poCmdMgr)
        {
            poInfo.mbsAddInName = "Demo AddIn";
            poInfo.mlRequiredVersionMajor = 17; //SOLIDWORKS PDM 2017 SP0

            poCmdMgr.AddCmd(TEST_CMD_ID, "Test Menu Command");
        }

        public void OnCmd(ref EdmCmd poCmd, ref Array ppoData)
        {
            if (poCmd.meCmdType == EdmCmdType.EdmCmd_Menu)
            {
                if (poCmd.mlCmdID == TEST_CMD_ID)
                {
                    (poCmd.mpoVault as IEdmVault10).MsgBox(0, "Hello World!");
                }
            }
        }
    }
}

File: solidworks-pdm-api/getting-started/add-ins/create/SampleAddIn.vb
Imports System.Runtime.InteropServices
Imports EdmLib

Namespace CodeStack

    <ComVisible(True)>
    <Guid("1B2547B5-F1F9-457D-941C-D25668619101")>
    Public Class SwPdmAddInVB
        Implements IEdmAddIn5

        Const CMD_ID As Integer = 1

        Public Sub GetAddInInfo(ByRef poInfo As EdmAddInInfo, poVault As IEdmVault5, poCmdMgr As IEdmCmdMgr5) _
        Implements IEdmAddIn5.GetAddInInfo

            poInfo.mbsAddInName = "Demo AddIn"
            poInfo.mlRequiredVersionMajor = 17

            poCmdMgr.AddCmd(CMD_ID, "Test Menu Command")

        End Sub

        Public Sub OnCmd(ByRef poCmd As EdmCmd, ByRef ppoData As Array) Implements IEdmAddIn5.OnCmd

            If poCmd.meCmdType = EdmCmdType.EdmCmd_Menu Then

                If poCmd.mlCmdID = CMD_ID Then
                    CType(poCmd.mpoVault, IEdmVault10).MsgBox(0, "Hello World!")

                End If

            End If

        End Sub
    End Class

End Namespace
